rust/core-lib/src/view.rs

```diff
@@ -125,8 +126,48 @@ impl View {
         self.cursor_col = col;
     }
 
-    pub fn get_cursor_col(&mut self) -> usize {
-        self.cursor_col
+    /// Toggles a caret at the given offset.
+    pub fn toggle_sel(&mut self, offset: usize) {
+        self.sel_dirty = true;
+        if !self.selection.regions_in_range(offset, offset).is_empty() {
+            self.selection.delete_range(offset, offset);
+            if !self.selection.is_empty() {
+                return;
+            }
+        }
+        let region = SelRegion {
+            start: offset,
+            end: offset,
+            // TODO: might want to set horiz to some meaningful value.
+            horiz: None,
```

Comment:
This is interesting. The previous way you calculated a `HorizPos` required access to the text, but the view doesn't seem to have that. Does that mean that the text needs to be passed as a parameter to `toggle_sel` somehow, should the `HorizPos` be passed in, or should calculating a `HorizPos` not rely on the text of the document?